lib/src/revset.rs

```diff
@@ -437,64 +438,162 @@ impl RevsetExpression {
         Rc::new(RevsetExpression::Difference(self.clone(), other.clone()))
     }
 
+    /// Commits that are in `self` but not in any of the `others`.
+    pub fn minus_all(
+        self: &Rc<RevsetExpression>,
+        others: &[Rc<RevsetExpression>],
+    ) -> Rc<RevsetExpression> {
+        if others.is_empty() {
+            self.clone()
+        } else {
+            RevsetExpression::minus(self, &RevsetExpression::union_all(others))
+        }
```

Comment:
nit: I'll inline this to caller.

`x & ~none()` could be rewritten to `x` at later pass, but apparently we don't have a rule for that right now.